util/loadAuthorCollections.ts

```diff
@@ -0,0 +1,29 @@
+import { promises as fs } from 'fs'
+
+import yaml from 'js-yaml'
+
+import AuthorCollection from './AuthorCollection'
+import { normalizeAuthorCollectionDefinition } from './validators'
+
+interface AuthorCollections {
+    authors: AuthorCollection[]
+}
+
+function returnAuthorCollection(authors: AuthorCollection[]): AuthorCollection[] {
+    let validatedAuthorCollections: AuthorCollection[] = []
+    authors.map(authorCollectionDefinition => {
+        const normalizedAuthorCollection = normalizeAuthorCollectionDefinition(authorCollectionDefinition)
+        validatedAuthorCollections = [ ...validatedAuthorCollections, normalizedAuthorCollection ]
+    })
+
+    return validatedAuthorCollections
+}
```

Comment:
Since `map` returns the modified array you could simplify the function:
```ts
function returnAuthorCollection(authors: AuthorCollection[]): AuthorCollection[] {
    return authors.map(authorCollectionDefinition => normalizeAuthorCollectionDefinition(authorCollectionDefinition))
}
```

Or further:

```ts
function returnAuthorCollection(authors: AuthorCollection[]): AuthorCollection[] {
    return authors.map(normalizeAuthorCollectionDefinition)
}
```

Reply:
Nice!